storage/innobase/srv/srv0start.cc

```diff
@@ -1781,11 +1795,14 @@ dberr_t srv_start(bool create_new_db)
 		ut_ad(high_level_read_only
 		      || srv_force_recovery < SRV_FORCE_NO_UNDO_LOG_SCAN);
 
-		if (!high_level_read_only
-		    && srv_sys_space.can_auto_shrink()) {
-			fsp_system_tablespace_truncate();
-			DBUG_EXECUTE_IF("crash_after_sys_truncate",
+		if (!high_level_read_only) {
+
+			if (srv_sys_space.can_auto_shrink()) {
+				fsp_system_tablespace_truncate(false);
+				DBUG_EXECUTE_IF(
+					"crash_after_sys_truncate",
 					return srv_init_abort(DB_ERROR););
+			}
```

Comment:
This looks like a nonfunctional change, replacing short-circuit evaluation with two nested `if`. Am I right that this hunk could be reverted without having any observable effect?